spec/index.bs

```diff
@@ -769,16 +771,42 @@ GPUBuffer includes GPUObjectBase;
     : <dfn>\[[state]]</dfn> of type [=buffer state=].
     ::
         The current state of the {{GPUBuffer}}.
+
+    : <dfn>\[[mapping]]</dfn> of type {{ArrayBuffer}}.
+    ::
+        The mapping for this {{GPUBuffer}}.
+
+    : <dfn>\[[mapping_promise]]</dfn> of type {{Promise}}.
+    ::
+        A promise for the mapping.
 </dl>
 
-Each {{GPUBuffer}} has a current <dfn dfn>buffer state</dfn> which is one of the following:
+Each {{GPUBuffer}} has a current <dfn dfn>buffer state</dfn> which is one of
+the following:
+
+ - "<dfn dfn for="buffer state">mapped for reading</dfn>" where the {{GPUBuffer}} is
+     available for CPU operations reading its content.
+ - "<dfn dfn for="buffer state">mapped for writing</dfn>" where the {{GPUBuffer}} is
+     available for CPU operations writing its content.
+ - "<dfn dfn for="buffer state">mapping pending for reading</dfn>" where the {{GPUBuffer}} is
+     being made available for reading its content.
+ - "<dfn dfn for="buffer state">mapping pending for writing</dfn>" where the {{GPUBuffer}} is
+     being made available for writing its content.
+ - "<dfn dfn for="buffer state">unmapped</dfn>" where the {{GPUBuffer}} is
+     available for GPU operations.
+ - "<dfn dfn for="buffer state">destroyed</dfn>" where the {{GPUBuffer}} is
+     no longer available for any operations except {{GPUBuffer/destroy}}.
 
- - "<dfn dfn for="buffer state">mapped</dfn>" where the {{GPUBuffer}} is available for CPU operations.
- - "<dfn dfn for="buffer state">unmapped</dfn>" where the {{GPUBuffer}} is available for GPU operations.
- - "<dfn dfn for="buffer state">destroyed</dfn>" where the {{GPUBuffer}} is no longer available for any operations except {{GPUBuffer/destroy}}.
+Note:
+{{GPUBuffer/[[size]]}} and {{GPUBuffer/[[usage]]}} are immutable once the
+{{GPUBuffer}} has been created.
 
 Note:
-{{GPUBuffer/[[size]]}} and {{GPUBuffer/[[usage]]}} are immutable once the {{GPUBuffer}} has been created.
+{{GPUBuffer}}'s {{GPUBuffer/[[state]]}} is a state machine where the states are
```

Comment:
do we have to say it's a state machine? I mean, it's not a machine, it's just a state.